feat: add chisel-wrapper script

rebornplusplus opened the pull request:
This wrapper script installs the slices and can generate an artificial status file similar to `/var/lib/dpkg/status`.

Basic usage:

	export CHISEL_DPKG_STATUS_FILE=/path/to/file
	chisel-wrapper [chisel-cut-OPTIONS] <slices..>

Please read through the script for detailed usage info.

woky commented on chisel-wrapper:
```diff
@@ -0,0 +1,53 @@
+set -eu
+
+CHISEL_BIN="${CHISEL_BIN:-"chisel"}"
+CHISEL_CACHE_DIR="${CHISEL_CACHE_DIR:-"$(mktemp -d)"}"
```
I understand that the point of `mktemp -d` here is to have an empty cache directory on each invocation so that we can see what appears in the cache. But since we create a new unique temporary directory and fill it with stuff on every invocation, we should at least clean it afterwards. And if we're going to clean it on every run, it's probably of no use to the caller of this script to configure where it is (maybe aside from putting it on the same filesystem).

So I suggest:
```
CHISEL_CACHE_DIR=$(mktemp -d)
cleanup() {
    rm -rf "$CHISEL_CACHE_DIR"
}
trap cleanup EXIT
```

rebornplusplus replied:
You have a good point. Cris mentioned it before as well. I kept `CHISEL_CACHE_DIR` in case anyone wanted to use the existing cache directory, and didn't clean up because the directory might be used later on. But since these are use cases that we are not immediately using, I suppose I can just remove the variable entirely and cleanup as you suggested.

woky replied:
> I kept CHISEL_CACHE_DIR in case anyone wanted to use the existing cache directory

But then the directory might exist and contain junk so you will have to clean it up first. Now you'd have two cases 1) a temporary directory that needs to be removed afterwards, 2) user provided directory that should be cleaned before. IMHO not worth the effort.

rebornplusplus replied:
Yeah, you are right. Added in https://github.com/canonical/rocks-toolbox/pull/1/commits/84640858c70ff3487b538ef482be1b48a21832aa. Thank you.

woky reviewed (approved):
Noice!

woky commented on chisel-wrapper:
```diff
@@ -0,0 +1,63 @@
+	done
+}
+
+if [ -z "$CHISEL_DPKG_STATUS_FILE" ] ; then
```
Since you have `set -u`, this will fail with variable unset error instead of going down into the if block. You can do either 

```
if [ -z "${CHISEL_DPKG_STATUS_FILE:-}" ]; then
```
or use Bashism
```
if [[ ! -v CHISEL_DPKG_STATUS_FILE ]]; then
```

woky commented on chisel-wrapper:
```diff
@@ -0,0 +1,63 @@
+trap cleanup EXIT
+
+print_error() {
+	echo "Error:" "$@" >> /dev/stderr
```
Why not just echo "Error:" "$@" >&2?

rebornplusplus replied:
Changed in https://github.com/canonical/rocks-toolbox/pull/1/commits/d0377af9488a00ac7455f4e3a256651ca7fffd7b. Thanks.

cjdcordeiro reviewed (changes requested):
sry, I had already approved but I'm having a change of heart wrt to the name of the script. now that we've decided to keep it (maybe for quite some time) and that maybe others (like msft) might be reusing this script, I think it deserves a better name.

`wrapper` means it will have more functionality on top of Chisel, but in this case it only has one, and doesn't really say what that is...

While one idea could simply be to rename this script to something a bit more specific, maybe the best and more future proof solution is the following:

 1. keep the file name
 2. add mandatory arguments to the script
     - the only one, for the moment, would be `--generate-dpkg-status` (bool)
 3. everything after `--` goes to `chisel`

Here's an example:

```bash
chisel-wrapper --generate-dpkg-status -- --release "ubuntu-$UBUNTU_RELEASE" --root /rootfs base-files_base
```

Without any args, it would just run chisel as is.

rebornplusplus commented:
Added the changes @cjdcordeiro suggested. Per our discussion later on, there is a minor change from the above comment:

> the only one, for the moment, would be --generate-dpkg-status (bool)

`--generate-dpkg-status` takes a required argument now, the path for the dpkg status file to be generated at.

Additionally, I added `-g` as a short form for `--generate-dpkg-status`, let me know if you want me to drop it.

cjdcordeiro commented on chisel-wrapper:
```diff
@@ -0,0 +1,105 @@
+
+	[OPTIONS]
+
+	  -g, --generate-dpkg-status <path>
```
y let's just drop `-g`. shorthands should be minimally intuitive and this one isn't cause it is a very tailored option.

cjdcordeiro reviewed (changes requested):
Awesome, thanks. Let's just drop `-g` and then merge. Did it pass your tests (we can add those later)?

rebornplusplus commented:
Well, the script passes the minimal local testing I did on it. All the options work well, it installs the file system and generates the status file successfully, and passes `shellcheck`.

cjdcordeiro reviewed (approved):
Thanks!
